src/main/scala/latis/input/HapiBinaryAdapter.scala

```diff
@@ -0,0 +1,120 @@
+package latis.input
+
+import java.net.URI
+import java.nio.{ByteBuffer, ByteOrder}
+
+import cats.effect.IO
+import fs2.Stream
+import latis.data.{Data, Sample}
+import latis.model.{DataType, Function, Scalar}
+
+/**
+ * Adapter for HAPI binary datasets.
+ */
+class HapiBinaryAdapter(model: DataType) extends StreamingAdapter[Iterator[Byte]] {
+  
+  val order = ByteOrder.LITTLE_ENDIAN //all numeric values are little endian (LSB)
+  
+  lazy val blockSize: Int = model.getScalars.map(bytesToRead).sum
+
+  /**
+   * Provides a Stream of records as Iterators of bytes.
+   */
+  def recordStream(uri: URI): Stream[IO, Iterator[Byte]] =
+    StreamSource.getStream(uri)
+      .chunkN(blockSize)
+      .map(_.iterator)
```

Comment:
I'm wondering if we can turn the iterator into a byte array here (i.e. Array[Byte] or ByteBuffer as the record) to avoid the `take` problem @lindholc mentioned. However, I do like the idea of trying scodec.

Reply:
That would certainly work.